File: vespalib/src/vespa/vespalib/net/tls/authorization_mode.cpp
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "authorization_mode.h"

#include <iostream>

namespace vespalib::net::tls {

const char* enum_name(AuthorizationMode mode) noexcept {
    switch (mode) {
    case AuthorizationMode::Disable: return "Disable";
    case AuthorizationMode::LogOnly: return "LogOnly";
    case AuthorizationMode::Enforce: return "Enforce";
    }
    abort();
}

std::ostream& operator<<(std::ostream& os, AuthorizationMode mode) {
    os << enum_name(mode);
    return os;
}

}

File: security-utils/src/main/java/com/yahoo/security/tls/AuthorizationMode.java
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.security.tls;

import java.util.Arrays;

/**
 * @author bjorncs
 */
public enum AuthorizationMode {
    DISABLE("disable"),
    LOG_ONLY("log_only"),
    ENFORCE("enforce");

    final String configValue;

    AuthorizationMode(String configValue) {
        this.configValue = configValue;
    }

    public String configValue() {
        return configValue;
    }

    /**
     * @return Default value when authorization mode is not explicitly specified
     */
    public static AuthorizationMode defaultValue() {
        return ENFORCE;
    }


    public static AuthorizationMode fromConfigValue(String configValue) {
        return Arrays.stream(values())
                .filter(v -> v.configValue.equals(configValue))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Unknown value: " + configValue));
    }
}
